### DISCUSSION TITLE
[typescript] noExplicitAny rule is dumb (black/white config)

### DISCUSSION THREAD

**Discussion by c0ncentus** [rome/tools#4167] [Suggestions]
Hi,
```Typescript
// making special things on opt
export function allToStr(data: any, opt?: object): string {
    return typeof data === "object"
        ? JSON.stringify(data)
        : data === "boolean"
            ? data === true
                ? "true"
                : "false"
            : typeof data === "number"
                ? data.toString()
                : typeof data === "undefined"
                    ? "undefined"
                    : typeof data === "string"
                        ? data
                        : data
}

export function Copy(data: any) { return <TwButtonFaRound 
        onClick={() => { copy(allToStr(data)) }}
        faPrefix="fa" 
        colorBg="gray"
        faName="copy"
 /> 
}
```
making error on playground.

I expect use "any" because expose as functionnality in typescript, it make sense on parameters only and very rare circonstance.

can you make an "any" mode : "authorizeOnParamater", "NoAny", "Any-way" mode  please ?

if Rome ban all any because it is "any" then i should do this (very ugly).
but at the same I want detect all "any" in my code <not in parameters> 

```Typescript

type Anyway =object|string|number|null|undefined|Function|boolean | object[]|string[]|number[]|null[]|undefined[]|Function[]|boolean[]

export function allToStr(data: Anyway , opt?: object): string {
    return typeof data === "object"
        ? JSON.stringify(data)
        : data === "boolean"
            ? data === true
                ? "true"
                : "false"
            : typeof data === "number"
                ? data.toString()
                : typeof data === "undefined"
                    ? "undefined"
                    : typeof data === "string"
                        ? data
                        : data
}

export function Copy(data: Anyway ) { return <TwButtonFaRound 
        onClick={() => { copy(allToStr(data)) }}
        faPrefix="fa" 
        colorBg="gray"
        faName="copy"
 /> 
}
}
```

When you convert .js to .ts, I used to have ~100 errors with gentle config, 
have "superStrict">"strict">"gentle">"no config" configuration will help to prioritize ...

Then it will better I think to add more shades/tolerance to "any" errors .

**Comment by Conaclos** [comment]
Hi :)

You can [disable the rule in the configuration file](https://docs.rome.tools/linter/#disable-a-lint-rule) or add a [`rome-ignore` comment](https://docs.rome.tools/linter/#ignoring-code) at the site where you use `any`.

**Reply by c0ncentus** [reply]
> Hi :)
> 
> You can [disable the rule in the configuration file](https://docs.rome.tools/linter/#disable-a-lint-rule) or add a [`rome-ignore` comment](https://docs.rome.tools/linter/#ignoring-code) at the site where you use `any`.

between doing a purge of "any" and simply doing like javascript thats very manichean, I only suggest consistently ignoring in function or method signatures.

Like typescript expose "ts-ignore", it is a great solution I know, but it's like linter ignore what you doing, That's strange ...

**Reply by Conaclos** [reply]
> I only suggest consistently ignoring in function or method signatures.

Could you use [unknown type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type) instead?

**Reply by c0ncentus** [reply]
Ok i will use that 👍
